### DISCUSSION TITLE
User session limiter added but user getting logout on Browser reload

### DISCUSSION THREAD

**Discussion by samir-kamble** [keycloak/keycloak#15918] [Q&A]
Hi All,

Thanks for this wonderful marvel. I am trying to setup User session delimiter on Keycloak version 18. I have done everything as per guidelines mentioned on following link

https://www.keycloak.org/docs/latest/server_admin/index.html#_user_session_limits

Also, setup everything correctly.  Referred following thread for setting up various possible options in configurations. 

https://github.com/keycloak/keycloak/issues/10077

Multiple session are working correctly. But when I reload browser. I get login screen back. But When I see my session in keycloak admin it is still there it is not killed.

Note: I am using AD authentication 

Attaching screenshot of my configuration. Please help.

![Capture](https://user-images.githubusercontent.com/12523579/206474596-72d4f39b-0468-4628-a4eb-6110826a6660.PNG)

**Comment by sschu** [comment]
You cannot have required and alternative authenticators at the same level. All of your top-level authenticators must be alternative or disabled. You can switch "Copy of browser forms" to alternative and remove the user session count limiter on the top level because new sessions will only be created after the username password form. You should set the session count limiter there to required.

**Reply by samir-kamble** [reply]
> You cannot have required and alternative authenticators at the same level. All of your top-level authenticators must be alternative or disabled. You can switch "Copy of browser forms" to alternative and remove the user session count limiter on the top level because new sessions will only be created after the username password form. You should set the session count limiter there to required.

Hi Thanks for Quick response, I tried your suggested config. 

![Capture](https://user-images.githubusercontent.com/12523579/206537529-ad976675-1b7a-41c9-a508-1dcbcdd1859e.PNG)

It gave me following error on login page

![tempsnip](https://user-images.githubusercontent.com/12523579/206537316-6f704d6d-880a-4c1c-8366-57ebb06be664.png)

**Reply by sschu** [reply]
Cookie and Copy of Browser Forms must be set to alternative.

**Reply by samir-kamble** [reply]
I have created following flow. Is it correct?  As I am getting 400 Bad request on refreshing page and Token is not getting generated.

![Capture](https://user-images.githubusercontent.com/12523579/206704416-a6a75c18-50f5-4ada-9cc8-1484d462b976.PNG)

**Reply by sschu** [reply]
The user session count limiter should only be after the Username Password form, not after the Cookie (because the latter does not create a session).

**Comment by samir-kamble** [comment]
Hi Sebastian,

Many thanks for your help much appreciated. You have guided perfectly. It is working as expected. 

Will test more and report any further issues being faced.

Regards,
Sam.

**Reply by hdaevbg** [reply]
Hi @samir-kamble can you please share the auth flow setup which is working? because i tried the same but still its not working and its logs out the user on both new and older sessions (different browsers). Here is the setup screenshot - 
![image](https://user-images.githubusercontent.com/82088001/210209508-69c58a8b-b3df-446f-aa08-46b4fa32711f.png)

**Reply by samir-kamble** [reply]
![Capture](https://user-images.githubusercontent.com/12523579/212737993-65907336-d48a-4afb-97b2-52caf3931835.PNG)

Follow above setup. It works well.
